add example support for floats and arrays

arvidfm opened the pull request:
This PR adds support for the `example` keyword for floats and arrays. For arrays, it expects a format like:

```go
MyArray []string `jsonschema:"example=A;B"`
```

arvidfm commented:
@samlown I've added tests now

samlown reviewed (approved):
Looks good! Thanks for this! It'd be great to fix `parseValue` return values if possible.

arvidfm commented:
@samlown Sorry for not having gotten around to this until now, but I've now addressed your comments. I adjusted the `Type == ""` case to accept any kind of value instead of just JSON objects, and added a corresponding test.
